src/char/include/ccharclient.h

```diff
@@ -24,13 +24,12 @@
 class CCharClient : public RoseCommon::CRoseClient {
  public:
   CCharClient();
-  CCharClient(tcp::socket _sock);
+  CCharClient(Core::INetwork* _sock);
```

Comment:
It's a raw pointer because I'm preserving the previous functionality. With using a smart ptr, I'm going to have to regression test everything.